extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -19,47 +20,170 @@ export abstract class ToolBase implements ITool {
 	abstract type: ToolType;
 	abstract homePage: string;
 	abstract autoInstallSupported: boolean;
-	public abstract install(): Promise<void>;
+	abstract readonly allInstallationCommands: { [key: string]: Command[] };
 	protected abstract getVersionFromOutput(output: string): SemVer | undefined;
```

Comment:
The key should be an OsType enum value

Reply:
The language supports only string or number types as keys. It is lame because OsType is essentially a mascaraded number

Reply:
Just use a map

Reply:
will do

Reply:
Done